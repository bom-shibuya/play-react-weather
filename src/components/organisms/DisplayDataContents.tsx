import React from 'react'
import { fetchWeather, WeathersState } from '../../store/weatherList'
import { AsyncDispatch } from '../../types'
import { Contents } from '../organisms'

// import styled from 'styled-components'
export interface IProps {
  cityName: string
  weathersState: WeathersState
  dispatch: AsyncDispatch
}

export const initialCityList = ['Tokyo', 'Osaka', 'Niigata']

export class DisplayDataContents extends React.Component<IProps> {
  public fetchInitialWeather(): Promise<void[]> {
    const { dispatch } = this.props
    return Promise.all(
      initialCityList.map(city =>
        dispatch(
          fetchWeather({
            city
          })
        )
      )
    )
  }
  public componentDidMount() {
    this.fetchInitialWeather()
  }
  public render() {
    const { cityName, weathersState } = this.props
    return <Contents cityName={cityName} weathersState={weathersState} />
  }
}
