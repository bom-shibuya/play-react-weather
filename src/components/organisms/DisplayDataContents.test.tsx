import axios from 'axios'
import { shallow, ShallowWrapper } from 'enzyme'
import React from 'react'
import configureStore, { MockStoreEnhanced } from 'redux-mock-store'
import thunk from 'redux-thunk'
import { DisplayDataContents, IProps } from './DisplayDataContents'

describe('DisplayDataContents component test', () => {
  const res = { data: { list: [] } }
  const dispatch = jest.fn()
  let wrapper: ShallowWrapper<IProps>
  let store: MockStoreEnhanced

  beforeEach(() => {
    const mockStore = configureStore([thunk])
    store = mockStore()
    const props: IProps = {
      cityName: 'Tokyo',
      dispatch: store.dispatch,
      weathersState: {
        isError: false,
        isFetching: false,
        lastUpdateAt: null,
        list: []
      }
    }
    wrapper = shallow(<DisplayDataContents {...props} />)
    store.clearActions()
  })
  test('component did mount should called dispatch with expected arguments', async () => {
    axios.get = jest.fn(() => Promise.resolve(res))

    await wrapper.instance().componentDidMount()
    expect(store.getActions()).toMatchSnapshot()
  })
})
