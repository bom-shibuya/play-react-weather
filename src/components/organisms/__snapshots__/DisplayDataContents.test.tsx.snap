// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayDataContents component test component did mount should called dispatch with expected arguments 1`] = `
Array [
  Object {
    "payload": Object {
      "city": "Tokyo",
    },
    "type": "FETCH_REQUEST",
  },
  Object {
    "payload": Object {
      "city": "Osaka",
    },
    "type": "FETCH_REQUEST",
  },
  Object {
    "payload": Object {
      "city": "Niigata",
    },
    "type": "FETCH_REQUEST",
  },
  Object {
    "payload": Object {
      "city": "Tokyo",
      "data": Array [],
    },
    "type": "FETCH_SUCCESS",
  },
  Object {
    "payload": Object {
      "city": "Osaka",
      "data": Array [],
    },
    "type": "FETCH_SUCCESS",
  },
  Object {
    "payload": Object {
      "city": "Niigata",
      "data": Array [],
    },
    "type": "FETCH_SUCCESS",
  },
]
`;
